python/pfs/drp/stella/fitPfsFluxReference.py

```diff
@@ -0,0 +1,978 @@
+            Configuration of the PFS top-end.
+        pfsMerged : `pfs.datamodel.pfsFiberArraySet.PfsFiberArraySet`
+            Typically an instance of PsfMerged.
+            It must have been whitened.
+        bbPdfs : `List[Optional[numpy.array]]`
+            `bbPdfs[i]`, if not None, is the probability distribution
+            of `pfsConfig.fiberId[i]` being of each model type,
+            determined by broad-band photometries.
+
+        Returns
+        -------
+        radialVelocities : `List[Optional[lsst.pipe.base.Struct]]`
+            Radial velocity for each fiber.
+            Each element, if not None, has `velocity` and `error`
+            as its member. See ``EstimateRadialVelocityTask``.
+        """
+        # Find the best model from broad bands.
+        # This model is used as the reference for cross-correlation calculation
+        bestModels = self.findRoughlyBestModel(bbPdfs)
+
+        radialVelocities = []
+        for spectrum, model in zip(fibers(pfsConfig, pfsMerged), bestModels):
+            if model.spectrum is None:
+                radialVelocities.append(None)
+                continue
+            modelSpectrum = convolveLsf(model.spectrum)
+            modelSpectrum = self.whitenSpectrum(modelSpectrum, mode="model")
+            radialVelocities.append(self.estimateRadialVelocity.run(spectrum, modelSpectrum))
+
+        return radialVelocities
+
+    def whitenSpectrum(self, spectra, *, mode):
+        """Whiten one or more spectra.
+
+        Parameters
+        ----------
+        spectra : `PfsSimpleSpectrum` or `PfsFiberArraySet`
+            spectra to whiten.
+        mode : `str`
+            "observed" or "model".
+            Whether ``spectra`` is from observation or from simulation.
+
+        Returns
+        -------
+        spectra : `PfsSimpleSpectrum` or `PfsFiberArraySet`
+            The same instance as the argument.
+        """
+        if mode == "observed":
+            fitContinuum = self.fitObsContinuum
+        if mode == "model":
+            fitContinuum = self.fitModelContinuum
+
+        # If `spectra` is actually a single spectrum,
+        # we put it into PfsFiberArraySet
+        if len(spectra.flux.shape) == 1:
+            original_spectrum = spectra
+            if not hasattr(spectra, "covar"):
+                spectra = promoteSimpleSpectrumToFiberArray(spectra, snr=400)
+            spectra = promoteFiberArrayToFiberArraySet(spectra, fiberId=1)
+        else:
+            original_spectrum = None
+
+        # This function actually works with `PfsFiberArraySet`
+        # nonetheless for its name.
+        # (PfsArm is a subclass of PfsFiberArraySet)
+        specset = SpectrumSet.fromPfsArm(spectra)
+
+        lines = ReferenceLineSet.fromRows([
+            ReferenceLine("Hbeta", 486.1, 1.0, ReferenceLineStatus.GOOD),
+            ReferenceLine("Halpha", 656.3, 1.0, ReferenceLineStatus.GOOD),
+        ])
```

Comment:
The values are inaccurage because they are in air. I am fixing them. There is no particular reason for the impreciseness. I will write more digits.

> should this list be more complete
I have not looked into it but Yamashita-san says that he wants to mask these two lines because they are so wide that they do not vanish with the median filter. He does not care about any other lines because they disappear with the median filter. It was I who decided to write the two line directly in the source code. I thought that a file with only two short lines would not deserve to exist separately.